lib/eslint/eslint.js

```diff
@@ -613,6 +613,13 @@ class ESLint {
             });
         }
 
+        // Check for the .eslintignore file, and warn if it's present.
+        if (existsSync(path.resolve(processedOptions.cwd, ".eslintignore"))) {
+            process.emitWarning(
+                "The \".eslintignore\" file is no longer supported. For an upgrade, switch to using the \"ignores\" property in a config object. See https://eslint.org/docs/latest/use/configure/migration-guide#ignoring-files for details.",
+                "ESLintIgnoreWarning"
```

Comment:
I would tweak the wording a little bit:

```suggestion
                "The \".eslintignore\" file is no longer supported. Switch to using the \"ignores\" property in \"eslint.config.js\": https://eslint.org/docs/latest/use/configure/migration-guide#ignoring-files",
                "ESLintIgnoreWarning"
```